File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/wrapper/entity/EntityLivingBase.java
package rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.entity;

import rbq.wtf.lycoris.client.wrapper.MapEnum;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapClass;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapField;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapMethod;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapperClass;
import rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.potion.Potion;

import java.lang.reflect.Field;
import java.lang.reflect.Method;

@WrapperClass(mcpName = "net.minecraft.entity.EntityLivingBase", targetMap = MapEnum.VANILLA189)
public class EntityLivingBase extends Entity {

    @WrapClass(mcpName = "net.minecraft.entity.EntityLivingBase", targetMap = MapEnum.VANILLA189)
    public static Class<?> EntityLivingBaseClass;
    @WrapField(mcpName = "moveStrafing", targetMap = MapEnum.VANILLA189)
    public static Field moveStrafing;
    @WrapField(mcpName = "moveForward", targetMap = MapEnum.VANILLA189)
    public static Field moveForward;
    @WrapMethod(mcpName = "isPotionActive", targetMap = MapEnum.VANILLA189)
    public static Method isPotionActive;

    public EntityLivingBase(Object obj) {
        super(obj);
    }

    public float getMoveForward() {
        return (float) getField(moveForward);
    }

    public float getMoveStrafing() {
        return (float) getField(moveStrafing);
    }

    public boolean isPotionActive(Potion p_isPotionActive_1_) {
        return (boolean) invoke(isPotionActive, p_isPotionActive_1_.getWrapObject());
    }
}

File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/wrapper/entity/EntityLivingBase.kt
package rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.entity

import rbq.wtf.lycoris.client.wrapper.MapEnum
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapClass
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapField
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapMethod
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapperClass
import rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.potion.Potion
import java.lang.reflect.Field
import java.lang.reflect.Method

@WrapperClass(mcpName = "net.minecraft.entity.EntityLivingBase", targetMap = MapEnum.VANILLA189)
open class EntityLivingBase(obj: Any) : Entity(obj) {
    val moveForward: Float
        get() = getField(Companion.moveForward) as Float
    val moveStrafing: Float
        get() = getField(Companion.moveStrafing) as Float

    fun isPotionActive(p_isPotionActive_1_: Potion): Boolean {
        return invoke(isPotionActive, p_isPotionActive_1_.wrapObject) as Boolean
    }

    companion object {
        @WrapClass(mcpName = "net.minecraft.entity.EntityLivingBase", targetMap = MapEnum.VANILLA189)
        lateinit var EntityLivingBaseClass: Class<*>

        @WrapField(mcpName = "moveStrafing", targetMap = MapEnum.VANILLA189)
        lateinit var moveStrafing: Field

        @WrapField(mcpName = "moveForward", targetMap = MapEnum.VANILLA189)
        lateinit var moveForward: Field

        @WrapMethod(mcpName = "isPotionActive", targetMap = MapEnum.VANILLA189)
        lateinit var isPotionActive: Method
    }
}